### PR TITLE
fix(neon): remove and dispose dagnling blocs after logout

### PR DESCRIPTION
**Pull request by Leptopoda**
fixes: #306

### CODE REVIEW

**Comment by provokateurin** [comment]
I think we need something similar for disposing the blocs of an app implementation when the list of available apps is updated and the server no longer has that app. It is very rare, but right now we don't dispose correctly in that case.

**Comment by Leptopoda** [comment]
I also added unit tests for `Disposable` and `AccountCache`

File: packages/neon/neon/lib/src/models/account_cache.dart [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+        return true;
+      }
+
+      return false;
+    });
+  }
+
+  /// Removes [account] and its associated value, if present, from the cache.
+  ///
+  ///If present the value associated with `account` is disposed.
```
Comment (provokateurin) [inline packages/neon/neon/lib/src/models/account_cache.dart:35]:
```suggestion
  /// If present the value associated with [account] is disposed.
```

File: packages/neon/neon/lib/src/models/account_cache.dart [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    final removed = _cache.remove(account?.id);
+    removed?.dispose();
+  }
+
+  /// The value for the given [account], or `null` if [account] is not in the cache.
+  T? operator [](final Account account) => _cache[account.id];
+
+  /// Associates the [account] with the given [value].
+  ///
+  /// If the account was already in the cache, its associated value is changed.
```
Comment (provokateurin) [inline packages/neon/neon/lib/src/models/account_cache.dart:46]:
```suggestion
  /// If the account was already in the cache, it's associated value is changed.
```

File: packages/neon/neon/test/account_cache_test.dart [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+      expect(cache[account0], isNull);
+
+      cache[account0] = disposable0;
+      expect(cache[account0], disposable0);
+
+      expect(cache[account0] ??= disposable1, disposable0);
+      expect(cache[account1] ??= disposable1, disposable1);
+    });
+
+    test('map functionality', () {
```
Comment (provokateurin) [inline packages/neon/neon/test/account_cache_test.dart:34]:
```suggestion
    test('update', () {
```

**Review by provokateurin — APPROVED** [review]

File: packages/neon/neon/lib/src/models/account_cache.dart (hunk 2 above)
Comment (Leptopoda) [reply packages/neon/neon/lib/src/models/account_cache.dart:46]: This is correct.
Its in the form of "sein/seine"
Not it is like "es ist"

**Review by provokateurin — APPROVED** [review]

**Pull request merged**